main.go

```diff
@@ -66,53 +70,112 @@ const (
 	webhookPort = 9443
 )
 
-func runPrometheusServer(metricsAddr string, tlsOptions common.SSPTLSOptions) error {
+func getPrometheusTLSConfig(ctx context.Context, certWatcher *certwatcher.CertWatcher) *tls.Config {
+	return &tls.Config{
+		// This callback executes on each client call returning a new config to be used
+		// please be aware that the APIServer is using http keepalive so this is going to
+		// be executed only after a while for fresh connections and not on existing ones
+		GetConfigForClient: func(_ *tls.ClientHelloInfo) (*tls.Config, error) {
+			cfg := &tls.Config{}
+			var err error
+			cfg.GetCertificate = certWatcher.GetCertificate
+
+			if controllers.TLSProfile == nil {
+				cfg.MinVersion = crypto.DefaultTLSVersion()
+				cfg.CipherSuites = nil
+				return cfg, nil
+			}
+
+			if controllers.TLSProfile.Type == ocpconfigv1.TLSProfileCustomType {
+				cfg.CipherSuites = common.CipherIDs(controllers.TLSProfile.Custom.Ciphers)
+				cfg.MinVersion, err = crypto.TLSVersion(string(controllers.TLSProfile.Custom.MinTLSVersion))
+				if err != nil {
+					return nil, err
+				}
+				return cfg, nil
+			}
+
+			cipherNames, minTypedTLSVersion := ocpconfigv1.TLSProfiles[controllers.TLSProfile.Type].Ciphers, ocpconfigv1.TLSProfiles[controllers.TLSProfile.Type].MinTLSVersion
+			cfg.CipherSuites = common.CipherIDs(cipherNames)
+			cfg.MinVersion, err = crypto.TLSVersion(string(minTypedTLSVersion))
+			if err != nil {
+				return nil, err
+			}
+			return cfg, nil
+		},
```

Comment:
Can we extract this to an external function? It is duplicated on `getTLSConfigFunc()`.

Reply:
Indeed